### ISSUE TITLE
teacup modifies attributes on rendering

### ISSUE THREAD

**Issue by loveencounterflow** [goodeggs/teacup#46]
When i do

```coffee
{render, ul, li} = require 'teacup'
d =
  foo:    'baz'
  name:   'roger'
  class:  'foo bar'
output = render ->
  ul ->
    li d, 'Bergamont'
    li d, 'Chamomile'
console.log output
```

i would expect the output to be 
```html
<ul><li class="foo bar" foo="baz" name="roger">Bergamont</li><li class="foo bar" foo="baz" foo="baz" name="roger">Chamomile</li></ul>
```

Instead, what i get is this:

```html
<ul><li class="foo bar" foo="baz" name="roger">Bergamont</li><li foo="baz" name="roger">Chamomile</li></ul>
```
where the class attribute has disappeared in the second tag. 

I guess this is caused by [line #100 of teacup.coffee](https://github.com/goodeggs/teacup/blob/master/src/teacup.coffee#L100) where the code deals with 'ordered attributes'. 

Both the concept of HTML having any kind of ordering in attributes and code that surreptitiously modifies arguments it is supposed to just render are bogus to me; i suggest to just delete lines 97 through 101 from the source.

**Comment by hurrymaplelad** [comment]
Good catch @loveencounterflow!

Attribute ordering is definitely only a developer convenience, added to make the DOM visually scannable.  If we can quantify a performance penalty, I'd be motivated to pull it out as a [plugin](https://github.com/goodeggs/teacup#plugins).  Otherwise, seems like a quick fix.

I'd change it to 
``` js
    obj = Object.create(obj)
    for name in @attrOrder when name of obj
      result += @renderAttr name, obj[name]
      obj[name] = false
```
Which should be handled correctly by `renderAttr` and doesn't incur the performance penalty of cloning.

PR welcome, ideally with test.

**Comment by asalant** [comment]
All you need is the clone of obj you suggest:

```
    obj = Object.create(obj)
    for name in @attrOrder when name of obj
      result += @renderAttr name, obj[name]
      delete obj[name]
```

**Comment by hurrymaplelad** [comment]
@asalant not sure about that, 
```coffee
o = {a:1}
p = Object.create o
delete p.a
console.log p.a 
> 1 

`delete` only modified "own" properties, don't think it can make prototype properties appear unset.
```

**Comment by loveencounterflow** [comment]
well, it's actually all about two attribute names, `id` and `class`. As such, it should be avoided to modify the attributes object in the first loop; in the second loop, it could be as simple as `continue if name in @attrOrder`. 

And yes, my temporary fix is indeed to clone the attributes object, but i think it's really horrible that i have to, not to mention that having to do so each time is both onerous and less efficient than just skipping those attributes in the second loop.

**Comment by loveencounterflow** [comment]
i've created a pull request: https://github.com/goodeggs/teacup/pull/47; it keeps the intended attribute ordering, but avoids to modify the attributes object; should be reasonably fast.

**Comment by hurrymaplelad** [comment]
Fixed by #47
